File: Repository/CursaDBRepository.cs
using Microsoft.Extensions.Logging;
using Motociclete.Domain;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Microsoft.Data.Sqlite;
using log4net;

namespace Motociclete.Repository
{
    public class CursaDBRepository : IRepositoryCursa
    {
        private static readonly ILog logger = LogManager.GetLogger(typeof(CursaDBRepository));
        private readonly DBUtils _dbUtils;

        public CursaDBRepository(IDictionary<string, string> properties)
        {
            _dbUtils = new DBUtils(properties);
            logger.Info("RepositoryDBUser initialized");
        }

        public long Insert(Cursa entity)
        {
            logger.Info("Inserting new Cursa");

            using (IDbConnection conn = _dbUtils.getConnection())
            {
                try
                {
                    if (conn is SqliteConnection sqliteConn)
                    {
                        string query = "INSERT INTO Curse (numarParticipanti, capMotor) VALUES (@numarParticipanti, @capMotor); SELECT last_insert_rowid();";

                        using (var cmd = new SqliteCommand(query, sqliteConn))
                        {
                            cmd.Parameters.AddWithValue("@numarParticipanti", entity.NumarParticipanti);
                            cmd.Parameters.AddWithValue("@capMotor", entity.CapMotor);

                            long id = (long)cmd.ExecuteScalar();
                            logger.Info($"Inserted new Cursa with id {id}");
                            return id;
                        }
                    }
                }
                catch (Exception ex)
                {
                    logger.Error("Error inserting Cursa: " + ex.Message);
                }
            }
            return -1;
        }


        public void UpdateById(long id, Cursa entity)
        {
            throw new NotImplementedException();
        }

        public void DeleteById(long id)
        {
            throw new NotImplementedException();
        }

        public Cursa GetById(long id)
        {
            logger.Info($"Fetching Cursa with id {id}");

            using (IDbConnection conn = _dbUtils.getConnection())
            {
                try
                {
                    if (conn is SqliteConnection sqliteConn)
                    {
                        string query = "SELECT * FROM Curse WHERE id = @id";

                        using (var cmd = new SqliteCommand(query, sqliteConn))
                        {
                            cmd.Parameters.AddWithValue("@id", id);
                            using (var reader = cmd.ExecuteReader())
                            {
                                if (reader.Read())
                                {
                                    return new Cursa(reader.GetInt64(0), reader.GetInt32(1), reader.GetInt32(2));
                                }
                            }
                        }
                    }
                }
                catch (Exception ex)
                {
                    logger.Error("Error fetching Cursa: " + ex.Message);
                }
            }
            return null;
        }

        public List<Cursa> GetAll()
        {
            logger.Info("Fetching all Curse");

            List<Cursa> curse = new List<Cursa>();

            using (IDbConnection conn = _dbUtils.getConnection())
            {
                try
                {
                    if (conn is SqliteConnection sqliteConn)
                    {
                        string query = "SELECT * FROM Curse";

                        using (var cmd = new SqliteCommand(query, sqliteConn))
                        {
                            using (var reader = cmd.ExecuteReader())
                            {
                                while (reader.Read())
                                {
                                    curse.Add(new Cursa(reader.GetInt64(0), reader.GetInt32(1), reader.GetInt32(2)));
                                }
                            }
                        }
                    }
                }
                catch (Exception ex)
                {
                    logger.Error("Error fetching all Curse: " + ex.Message);
                }
            }
            return curse;
        }

    }

}

File: src/main/java/org/example/Repository/CursaDBRepository.java
package org.example.Repository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.example.Domain.Cursa;
import org.example.Domain.Participant;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.Properties;

import java.util.List;

public class CursaDBRepository implements CursaRepositoryInterface{
    private static final Logger logger = LogManager.getLogger(CursaDBRepository.class);
    private JdbcUtils dbUtils;

    public CursaDBRepository(Properties properties) {
        logger.info("Initializing AngajatDBRepository with properties: {}", properties);
        dbUtils = new JdbcUtils(properties);
    }
    @Override
    public Long insert(Cursa entity) {
        logger.traceEntry("Inserting new Participant");
        Connection conn = dbUtils.getConnection();

        try (PreparedStatement prepStmt = conn.prepareStatement("INSERT INTO Participant(numarParticipanti,capMotor) VALUES (?,?)")) {
            prepStmt.setInt(1, entity.getNumarParticipanti());
            prepStmt.setInt(2, entity.getCapMotor());
            prepStmt.executeUpdate();
            ResultSet rs = prepStmt.getGeneratedKeys();
            if (rs.next()) {
                logger.traceExit("Inserted new Participant with id{}", rs.getLong(1));
                return rs.getLong(1);
            }
        } catch (SQLException ex) {
            logger.error("error at insertion: " + ex);
            System.out.println("DB error: " + ex);
        }
        return null;
    }

    @Override
    public void updateById(Long aLong, Cursa entity) {

    }

    @Override
    public void deleteById(Long aLong) {

    }

    @Override
    public Cursa getById(Long aLong) {
        return null;
    }

    @Override
    public List<Cursa> getAll() {
        logger.info("starting getAll Curse");
        List<Cursa> curse = new ArrayList<>();
        Connection conn = dbUtils.getConnection();
        try (PreparedStatement prepStmt = conn.prepareStatement("SELECT * FROM Curse")){
            ResultSet rs = prepStmt.executeQuery();
            while (rs.next()) {
                curse.add(new Cursa(rs.getLong(1),rs.getInt(2),rs.getInt(3)));
            }
            logger.info("getAll Curse terminat");
            return curse;
        } catch (Exception e) {
            logger.error("eraore la getAll Curse " + e.getMessage());
        }
        return null;
    }
}
